hooks/pre_validations/10-postgresql_checkpoint_segments.rb

```diff
@@ -0,0 +1,17 @@
+custom_hiera = YAML.load_file('/etc/foreman-installer/custom-hiera.yaml')
+
+if custom_hiera.key?('postgresql::server::config_entries') &&
+     custom_hiera['postgresql::server::config_entries'].key?('checkpoint_segments')
+
+  message= <<~HEREDOC
+  ERROR:
+    checkpoint_segments tuning found in /etc/foreman-installer/custom-hiera.yaml
+    This tuning option is no longer valid in PostgreSQL 12+
+    Please remove this from the following locations and then re-run the installer:
+      - /etc/foreman-installer/custom-hiera.yaml and re-run the installer
+      - /var/opt/rh/rh-postgresql12/lib/pgsql/data/postgresql.conf
```

Comment:
This path is only relevant for EL7. It's unlikely other users will hit this, but worth to consider.

Users who have not yet upgraded will not have this file either since this is at the pre_validations stage. That might confuse users. Perhaps we should check if the file exists before suggesting this?